src/goto-harness/memory_snapshot_harness_generator.cpp

```diff
@@ -0,0 +1,310 @@
+/******************************************************************\
+
+Module: Harness to initialise memory from memory snapshot
+
+Author: Daniel Poetzl
+
+\******************************************************************/
+
+#include "memory_snapshot_harness_generator.h"
+
+#include <goto-programs/goto_convert.h>
+
+#include <json/json_parser.h>
+
+#include <json-symtab-language/json_symbol_table.h>
+
+#include <util/exception_utils.h>
+#include <util/fresh_symbol.h>
+#include <util/message.h>
+#include <util/string2int.h>
+#include <util/string_utils.h>
+#include <util/symbol_table.h>
+
+#include "goto_harness_generator_factory.h"
+
+void memory_snapshot_harness_generatort::handle_option(
+  const std::string &option,
+  const std::list<std::string> &values)
+{
+  if(option == "memory-snapshot")
+  {
+    memory_snapshot = require_exactly_one_value(option, values);
+  }
+  else if(option == "initial-location")
+  {
+    const std::string initial_location =
+      require_exactly_one_value(option, values);
+
+    std::vector<std::string> start;
+    split_string(initial_location, ':', start, true);
+
+    if(
+      start.size() == 0 || (start.size() >= 1 && start.front().empty()) ||
```

Comment:
1) Use `start.empy()` instead of `size() == 0`
2) `start.size() >= 1` is trivially true here

Reply:
Done.